File: python/commands/valid_commands/network_tag.py
from commands.valid_commands.base import CommandBase


class NetworkTagCmd(CommandBase):
    TAG: str = '0x1'
    NAME: str = 'метка сети'
    MAX_BYTES: int = 255

    def validate(self):
        return self.is_within_max_bytes()

    def is_within_max_bytes(self) -> bool:
        byte_length = len(self.string.encode('utf-8'))
        return byte_length <= self.MAX_BYTES

File: rust/commands/valid_commands/network_tag.rs
const TAG: &str = "0x1";
const NAME: &str = "метка сети";
const MAX_BYTES: usize = 255;

pub struct NetworkTagCmd {
    pub value: String,
}

impl NetworkTagCmd {
    pub fn new(value: String) -> Self {
        NetworkTagCmd { value }
    }

    pub fn validate(&self) -> bool {
        self.value.len() <= MAX_BYTES
    }

    pub fn name(&self) -> &str {
        NAME
    }

    pub fn tag(&self) -> &str {
        TAG
    }
}